### PR TITLE
upgrade the min version of Go from the docs 

### PR DESCRIPTION
**Pull request by chinchaun**
Signed-off-by: Jean Delannoy ([jdelannoy@gmail.com](mailto:jdelannoy@gmail.com))

Issue https://github.com/firecracker-microvm/firecracker-containerd/issues/731

Description of changes:
Update the min required Go version from the getting-started and quickstart guides

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: docker-credential-mmds/go.mod [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module github.com/firecracker-microvm/firecracker-containerd/docker-credential-mmds
 
-go 1.16
+go 1.17
```
Comment (swagatbora90) [inline docker-credential-mmds/go.mod:3]: you will probably need to run `go mod tidy` in all the places where you are updating go.mod.

There is a script that automates this in [tidy.sh](https://github.com/firecracker-microvm/firecracker-containerd/blob/main/tools/tidy.sh). Running `make tidy` also executes this script https://github.com/firecracker-microvm/firecracker-containerd/blob/main/Makefile#L135-L136

Comment (chinchaun) [reply docker-credential-mmds/go.mod:3]: thanks for the clarification @swagatbora90, I just updated the PR

**Comment by swagatbora90** [comment]
Some CI failures should be fixed now. Can you rebase and rerun CI for this PR.

**Comment by chinchaun** [comment]
> Some CI failures should be fixed now. Can you rebase and rerun CI for this PR.

Still broken :(

File: examples/cmd/remote-snapshotter/go.mod [hunk 2]
```diff
@@ -1,11 +1,46 @@
...
+	github.com/golang/groupcache v0.0.0-20210331224755-41bb18bfe9da // indirect
+	github.com/golang/protobuf v1.5.2 // indirect
+	github.com/google/uuid v1.3.0 // indirect
+	github.com/klauspost/compress v1.15.6 // indirect
+	github.com/moby/locker v1.0.1 // indirect
+	github.com/moby/sys/mountinfo v0.6.2 // indirect
+	github.com/moby/sys/signal v0.7.0 // indirect
+	github.com/opencontainers/go-digest v1.0.0 // indirect
+	github.com/opencontainers/image-spec v1.0.3-0.20211202183452-c5a74bcca799 // indirect
+	github.com/opencontainers/runc v1.1.3 // indirect
```
Comment (fangn2) [inline examples/cmd/remote-snapshotter/go.mod:31]: This had been upgraded to v1.1.5, probably you need to 
```
cd examples/cmd/remote-snapshotter
go mod tidy
```

**Review by fangn2 — CHANGES_REQUESTED** [review]
Your commit history seems not quite right. 
Now you are adding two more commits to your branch from main branch instead of rebasing. 
```
$ git log
commit 2c5a9b0443b1430432b1ea7b25504521d9181af9 (HEAD -> 731-update-go-version)
Author: Tony Fang <nhfang@amazon.com>
Date:   Tue Mar 28 21:41:20 2023 +0000

    Update runc to 1.1.5 from 1.1.3

    Signed-off-by: Tony Fang <nhfang@amazon.com>

commit 301678412ee284ab29271df0a461f75609e57a8d
Author: Vaishnavi Vejella <vvejella@amazon.com>
Date:   Wed Jan 11 19:55:04 2023 +0000

    Specify arch changes to dockerfile

    - Changed CTRD, CRITEST to install required binaries for X86_64 and ARM

    Signed-off-by: Vaishnavi Vejella <vvejella@amazon.com>

commit 2c3aaafd3e9a8861b29b0f0b502bdaa654ff283f
Author: Jean Delannoy <jjdelannoy@hotmail.com>
Date:   Fri Mar 17 14:23:23 2023 -0300

    make tidy Signed-off-by: Jean Delannoy (jdelannoy@gmail.com)

    Signed-off-by: Jean Delannoy <jjdelannoy@hotmail.com>
```

You probably need to sync your fork's main branch with the main repo first if you haven't done so.
Then locally switch to main branch, pull latest to it. 
Checkout to the feature branch(may need git reset to remove the latest two commits not related to your changes, based on your commit log), then
`git rebase main`

File: examples/cmd/remote-snapshotter/go.mod (hunk 2 above)
Comment (chinchaun) [reply examples/cmd/remote-snapshotter/go.mod:31]: thanks for head ups!

**Comment by fangn2** [comment]
Thank you very much for updating the PR!  Looks good to me.

P.S. CI is failing due to known issues which have been merged into main. 
You need to rebase main again to fix that.

**Comment by chinchaun** [comment]
> Thank you very much for updating the PR! Looks good to me.
> 
> P.S. CI is failing due to known issues which have been merged into main. You need to rebase main again to fix that.

hmmm if you mean the commit of this PR: https://github.com/firecracker-microvm/firecracker-containerd/pull/744
is included in my rebase

File: examples/cmd/remote-snapshotter/go.mod (hunk 2 above)
Comment (fangn2) [reply examples/cmd/remote-snapshotter/go.mod:31]: Have you done a `go mod tidy` here? 
Seems the CI is failing for this.

**Review by fangn2 — COMMENTED** [review]
You are right, sorry I saw all builds were failing and assumed the same CI issue.

A closer look I found below error for all builds.  Seems you forgot to `go mod tidy` in the `examples/cmd/remote-snapshotter` directory. 
``` 
go build -o remote-snapshotter ./remote_snapshotter.go
go: downloading github.com/containerd/stargz-snapshotter v0.11.3
go: downloading google.golang.org/grpc v1.47.0
go: downloading google.golang.org/genproto v0.0.0-20220617124728-180714bec0ad
go: github.com/opencontainers/runc@v1.1.3: missing go.sum entry; to add it:
	go mod download github.com/opencontainers/runc
make[2]: *** [Makefile:22: remote-snapshotter] Error 1
make[1]: *** [Makefile:31: remote-snapshotter] Error 2
```

**Review by fangn2 — APPROVED** [review]

**Comment by fangn2** [comment]
Changes look good and CI is happy.  

Only minor is: maybe better to squash fours commits into one to make git history clearer, since the fours commits altogether do the same thing.

**Pull request merged**